modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/podcast/EpisodeManagerImpl.kt

```diff
@@ -954,12 +955,19 @@ class EpisodeManagerImpl @Inject constructor(
 
     // Playback manager is only optional for UI tests. Should never be optional in the app but can't work out
     // another way without mocking a lot of stuff.
-    override fun archiveAllInList(episodes: List<PodcastEpisode>, playbackManager: PlaybackManager?) {
-        episodes.filter { !it.isArchived }.chunked(500).forEach { chunked ->
-            episodeDao.archiveAllInList(chunked.map { it.uuid }, System.currentTimeMillis())
-            playbackManager?.let { playbackManager ->
-                chunked.forEach {
-                    cleanUpEpisode(it, playbackManager)
+    override fun archiveAllInList(
+        episodes: List<PodcastEpisode>,
+        playbackManager: PlaybackManager?
+    ) {
+        launch(Dispatchers.IO) {
+            appDatabase.withTransaction {
+                episodes.filter { !it.isArchived }.chunked(500).forEach { chunked ->
```

Comment:
With this all wrapped in a transaction, I don't think there's any need to keep chunking this list. In my testing, there was no noticeable difference with and without the chunks. I left them in for now to minimize the amount of change in this PR since it is targeting the beta branch right before release.